# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/cairo/cairo/src/cairo-png.c
# Commit: 3cdfb39376b7
# Full Hash: 3cdfb39376b736597fe658a322389117686b7e82
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-05-07 09:13:07
# Regressor Bug: 1892913
# File Overlap Count: 3
# Description:
#   Bug 1892913 - patch 1 - Update cairo source to 1.18.0 r=gfx-reviewers,lsalzman
#   
#   Wholesale replacement of files in gfx/cairo/cairo with contents of the 1.18.0 release tarball,
#   omitting subdirectories that are not part of the core library build.
#   
# ==============================================================================

diff -r 9daa351f875b -r 3cdfb39376b7 gfx/cairo/cairo/src/cairo-png.c
--- a/gfx/cairo/cairo/src/cairo-png.c	Mon May 06 19:22:35 2024 +0000
+++ b/gfx/cairo/cairo/src/cairo-png.c	Mon May 06 19:25:12 2024 +0000
@@ -407,6 +407,14 @@
      */
     png_write_info (png, info);
 
+#ifndef WORDS_BIGENDIAN
+    /* libpng treats 16-bit data as big-endian by default. Swapping the
+     * byte-order on little endian ensures the native-endian data can be
+     * provided to png_write_image. This does not affect 8-bit data.
+     */
+    png_set_swap (png);
+#endif
+
     if (png_color_type == PNG_COLOR_TYPE_RGB_ALPHA) {
 	if (clone->format != CAIRO_FORMAT_RGBA128F)
 	    png_set_write_user_transform_fn (png, unpremultiply_data);
@@ -561,7 +569,6 @@
 
     return write_png (surface, stream_write_func, &png_closure);
 }
-slim_hidden_def (cairo_surface_write_to_png_stream);
 
 static inline int
 multiply_alpha (int alpha, int color)
@@ -702,6 +709,14 @@
 
     png_read_info (png, info);
 
+#ifndef WORDS_BIGENDIAN
+    /* libpng treats 16-bit data as big-endian by default. Swapping the
+     * byte-order on little endian ensures the native-endian data can be
+     * provided to png_read_image. This does not affect 8-bit data.
+     */
+    png_set_swap (png);
+#endif
+
     png_get_IHDR (png, info,
                   &png_width, &png_height, &depth,
                   &color_type, &interlace, NULL, NULL);